lib/p2p/Peer.ts

```diff
@@ -265,16 +270,45 @@ class Peer extends EventEmitter {
         resolve();
       };
 
-      const onTimeout = () => {
+      const onError = (err: Error) => {
         cleanup();
-        reject(new Error('Connection timed out.'));
-      };
 
-      this.socket!.once('connect', onConnect);
+        if (!retry) {
+          this.close();
```

Comment:
`this.close()` is not just for closing the connection, but to close the peer object entirely. 
It was indeed previously called from the `close` event which would trigger here, but I changed it to be called from there only if connection was already established. It was necessary for allowing retries.

Reply:
OK that makes sense. I think maybe the only minor consideration then would be to only emit the 'close' event from `peer.close()` if the connection had already been successfully opened.

Reply:
I'd rather leave it for now, since closing is not merely about the connection closing, and this event case might have some usage (specially if we'll want to track non-connected peers).